### ISSUE TITLE
confirmation/recieve behavior (especially for HW wallets)

### ISSUE THREAD

**Issue by My1** [Nault/Nault#170]
currently when recieving nano it immediately spawns a signing request to the ledger device out of nowhere without any explanation, maybe rather spawn a notification and just say to start the recieve process, and for the initial recieve, explain about representatives and maybe let the user immediately choose one or at least tell them what they are signing, as knowledge is the anchorpoint behind signing stuff.

an idea for also not making auto-recievers sad would be for the following:

1) initial recieve block always explained and manually triggered (even a short "you need a representative, we chose one for you, you can change it later if you want", along with the details of what the user is about to confirm)
2) in the settings for the recieve mode maybe add a checkbox, "initiate recieve manually in the above order" or "full auto recieve" depending on which direction would be better, which basically would switch between a 1-click batch recieve and a fully automatic recieve
3) when the hw wallet asks the user to confirm, show the user what is supposed to be confirmed.

**Comment by Joohansson** [comment]
I agree with some kind of message on the initial receive could be useful, however only possible in Nault. Can't be shown on the device unless the developer of the nano app can remake it.

I'm not sure we need another level of semi-auto receive. If you don't want it to be automatic it's easy to change in the settings to manual and for every new transaction, you get a notification along with a "new" icon plus an amount shown in the dropdown for the sidebar account selector. If you are in the receive screen they will automatically be populated there as well and you only need to press a button.

**Comment by My1** [comment]
well in nault was my intention (hence posting here). on the device there's enough going on in a check already so the big explanation should be left to the big computer.

the thing I think that's slightly bad with autorecieve is how it literally triggers even if nault is in background, and that can get very weird very fast especially considering it's the default. if using it as a pure software wallet there obviously isnt a need for this
manual on the other hand seems more like that you would have to recieve every transaction one by one, which might not be awesome either if you are going to recieve a lot.

**Comment by Joohansson** [comment]
Ok got it. A message in Nault telling you that you should confirm on the device to receive xx Nano from abc makes sense. I think we should have it.

**Comment by My1** [comment]
btw can nault ask the hw wallet whether it has auto-recieve on? because if yes the difference between "click to start batch recieve" and full auto could be precisely that.

> Ok got it. A message in Nault telling you that you should confirm on the device to receive xx Nano from abc makes sense

as well as a little extra info on reps on initial recieve

**Comment by Joohansson** [comment]
> btw can nault ask the hw wallet whether it has auto-recieve on? because if yes the difference between "click to start batch recieve" and full auto could be precisely that.

Don't know yet but it's not really the same thing IMO. In the device it means to skip on-device verification (which users have been reporting doesn't work always) while the auto in Nault is to receive everything without verification. The manual setting in Nault is meant to give more control in case you receive 100 transactions and just want to confirm 5 of them. I think they should continue to be completely independent of each other.

**Comment by My1** [comment]
I know its not exactly the same thing, but a full auto recieve would work a lot better on a device which has a skip for recieve verification on since it doesnt trigger a verification or anything out of nowhere.

the idea was just that if device verification is on that it would maybe wouldnt do a big verify of what you recieve in the nault but at least ask the user to start the verification on device.

that auto-recieve on the device side doesnt always work is something I havent been aware but a certainly interesting info
